Distribution/Server/Features/Tags.hs

```diff
@@ -195,20 +230,78 @@ tagsFeature CoreFeature{ queryGetPackageIndex
         pkgMap <- liftM packageTags $ queryState tagsState GetPackageTags
         return $ Map.fromDistinctAscList . map (\pkg -> (pkg, Map.findWithDefault Set.empty pkg pkgMap)) $ Set.toList pkgs
 
+    mergeTags :: Tag -> ServerPartE ()
+    mergeTags deprTag = do
+        tags <- optional $ look "tags"
```

Comment:
These front-end bits don't really belong here. `mergeTags` should accept the tags as function args and the html layer should do the lookup and parsing.